packages/component-base/src/virtualizer-iron-list-adapter.js

```diff
@@ -304,6 +304,13 @@ export class IronListAdapter {
 
     super._scrollHandler();
 
+    if (this._physicalCount !== 0) {
+      // After running super._scrollHandler, fix _virtualStart to workaround an iron-list issue.
+      // See https://github.com/vaadin/web-components/issues/1691
+      const reusables = this._getReusables(true);
+      this._virtualStart += reusables.indexes.length;
+    }
```

Comment:
Just calling [`this._render()`](https://github.com/vaadin/web-components/blob/e2bd6434a5b3278749871c876585cf886ce18715/packages/component-base/src/iron-list-core.js#L536-L542) here would be enough to fix the issue, but to avoid unnecessary overhead, let's just run the minimal necessary subset of the function.